### PR TITLE
Do not pull overlay data when overlay is disable

### PR DESCRIPTION
**Pull request by janisz**
## High-level description

Currently, we pull overlay endpoints to diagnostic bundle

* `:5050/overlay-master/state`
* `:5051/overlay-agent/overlay`

This PR will remove these endpoints when the overlay is [disabled](https://docs.mesosphere.com/1.12/installing/production/advanced-configuration/configuration-reference/#dcos-overlay-enable).

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS_OSS-4250](https://jira.mesosphere.com/browse/DCOS_OSS-4250) Avoid failure when overlay and related endpoints are not available

## Checklist for all PRs

  - [ ] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [ ] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by janisz** [comment]
@GoelDeepak @bernadinm @klueska @jgehrcke Can you take a look?

**Comment by janisz** [comment]
@mesosphere-mergebot bump-ee

**Comment by janisz** [comment]
@mesosphere-mergebot bump-ee

File: gen/dcos-config.yaml [hunk 1]
```diff
@@ -701,6 +701,20 @@ package:
     content: |
         {
           "HTTPEndpoints": [
+            {% switch dcos_overlay_enable %}
+             {% case "true" %}
```
Comment (klueska) [inline gen/dcos-config.yaml]: The spacing seems weird here. Is a single space indent intended?

Comment (branden) [reply gen/dcos-config.yaml]: To match the style of the rest of this template, the `switch` and `case` statements here, and the `case` and `endswitch` below, should be completely de-indented to the beginning of their respective lines, with no preceding whitespace.

**Comment by janisz** [comment]
@mesosphere-mergebot bump-ee

**Comment by janisz** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/aws/onprem/static https://jira.mesosphere.com/browse/DCOS_OSS-2115

**Comment by janisz** [comment]
@branden It's fixed now

**Comment by janisz** [comment]
@branden @orsenthil @philipnrmn Can you override VIP failing test?

**Comment by janisz** [comment]
@mesosphere-mergebot bump-ee

**Comment by orsenthil** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/azure/arm DCOS-41675

**Pull request merged**